pysmt/test/test_string.py

```diff
@@ -0,0 +1,309 @@
+#
+# This file is part of pySMT.
+#
+#   Copyright 2015 Andrea Micheli and Marco Gario
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+from pysmt.test import TestCase, skipIfNoSolverForLogic, main
+from pysmt.shortcuts import *
+from pysmt.printers import  HRPrinter
+from pysmt.typing import INT,STRING
+from pysmt.logics import QF_SLIA
+
+class TestSTRING(TestCase):
+    
+    
+    @skipIfNoSolverForLogic(QF_SLIA)
+    def test_str_length(self):
+        s1 = Symbol("s1",STRING)
+        s2 = Symbol("s2",STRING)
+        solver = Solver(logic=QF_SLIA)
+        solver.add_assertion(  Not( Implies( Equals(s1,s2),
+                                             Equals(str_length(s2), str_length(s1))
+                                            )
+                                   )
+                             )
+        res = solver.solve()
+        if ( res):
+            print solver.get_model()
+        assert not res, "Was expecting  FALSE" 
+        return
+    
+    
+    @skipIfNoSolverForLogic(QF_SLIA)
+    def test_str_concat(self):
+        s1 = Symbol("s1",STRING)
+        s2 = Symbol("s2",STRING)
+        solver = Solver(logic=QF_SLIA)
+        solver.add_assertion( Not (And (
+                                   GE(
+                                      str_length(str_concat(s1,s2)), 
+                                      str_length(s1)
+                                    ), 
+                                   GE(
+                                      str_length(str_concat(s1,s2)), 
+                                      str_length(s2)
+                                      )
+                                   )
+                                   )
+                             )
+        res = solver.solve()
+        if ( res):
+            print solver.get_model()
+        assert not res, "Was expecting  FALSE" 
+        return
+    
+    
+    @skipIfNoSolverForLogic(QF_SLIA)
+    def test_str_contains(self):
+        s1 = Symbol("s1",STRING)
+        s2 = Symbol("s2",STRING)
+        solver = Solver(logic=QF_SLIA)
+        solver.set_option('strings-exp','true')
```

Comment:
This option is really specific only to CVC4 and, btw, it is already set within the solver. No need to have this here.